structureprovider/factory.go

```diff
@@ -0,0 +1,70 @@
+package structureprovider
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/USACE/go-consequences/consequences"
+	"github.com/USACE/go-consequences/geography"
+)
+
+const (
+	Provider_NSI   string = "Provider_NSI"
+	Provider_Local        = "Provider_Local"
+)
```

Comment:
How about string enums? It's easier to generate custom errors

Reply:
that is great